File: object-c/objetive-c/objetive-c/Player.m
//
//  Player.m
//  objetive-c
//
//  Created by Jian Su on 3/7/15.
//  Copyright (c) 2015 Jian Su. All rights reserved.
//

#import "Player.h"

@implementation Player

- (instancetype)init
{
    self = [super init];
    if (self) {
        _employeeID = 5000;
        _employeeName = @"Chris Su";
    }
    return self;
}

- (instancetype)initWithID:(int)employeeID initWithName:(NSString *)employeeName
{
    self = [super init];
    if (self) {
        _employeeID = employeeID;
        _employeeName = employeeName;
    }
    return self;
}

@end

File: swift/rpgoop/rpgoop/Player.swift
//
//  Player.swift
//  rpgoop
//
//  Created by Jian Su on 2015-10-25.
//  Copyright © 2015 Jian Su. All rights reserved.
//

import Foundation

class Player: Character {
    private var _name = "Player"
    
    var name:String {
        get {
            return _name
        }
    }
    
    private var _inventory = [String]()
    
    var inventory: [String] {
        return _inventory
    }
    
    func addItemToInventory(item: String) {
        _inventory.append(item)
    }
    
    convenience init(name:String, hp:Int, attackPwr:Int) {
        self.init(startingHp: hp, attackPwr: attackPwr)
        _name = name
    }
}